packages/solana/src/staker.ts

```diff
@@ -357,31 +358,37 @@ export class SolanaStaker {
     signer: Signer
     signerAddress: string
     tx: SolanaTransaction
-  }): Promise<{ signedTx: Transaction }> {
+  }): Promise<{ signedTx: VersionedTransaction }> {
     const connection = this.getConnection()
     const { signer, signerAddress, tx } = params
-    const { blockhash, lastValidBlockHeight } = await connection.getLatestBlockhash()
+    const { blockhash } = await connection.getLatestBlockhash()
+    const versionedTransaction = new VersionedTransaction(
+      new TransactionMessage({
+        payerKey: new PublicKey(signerAddress),
+        recentBlockhash: blockhash,
+        instructions: tx.tx.instructions
+      }).compileToV0Message()
+    )
 
-    // the owner of the stake account is the fee payer
-    tx.tx.feePayer = new PublicKey(signerAddress)
+    const serializedMessage = versionedTransaction.message.serialize()
+    let message: string = ''
+    if (Buffer.isBuffer(serializedMessage)) {
```

Comment:
I wonder in what conditions would the selializeMessage be "the other" type. Isn't serialize() returning always the same thing/type?

Reply:
this is the method:
 `serialize(): Buffer | Uint8Array<ArrayBufferLike>`
Not sure under which condition it will return the other type, but TS was screaming, so had to go with the ugly if/else